packages/esm-service-queues-app/src/hooks/useQueues.ts

```diff
@@ -0,0 +1,16 @@
+import { openmrsFetch } from '@openmrs/esm-framework';
+import useSWRImmutable from 'swr/immutable';
+import { type Queue } from '../types';
+
+export function useQueues(locationUuid?: string) {
+  const customRepresentation =
+    'custom:(uuid,display,name,description,allowedPriorities:(uuid,display),allowedStatuses:(uuid,display))';
+  const apiUrl = `/ws/rest/v1/queue?v=${customRepresentation}` + (locationUuid ? `&location=${locationUuid}` : '');
+
+  const { data, ...rest } = useSWRImmutable<{ data: { results: Array<Queue> } }, Error>(apiUrl, openmrsFetch);
```

Comment:
```suggestion
  const { data, ...rest } = useSWRImmutable<{ data: { results: Array<Queue> } }, Error>(locationUuid ? apiUrl : null, openmrsFetch);
```

Tip: You can make data fetching [conditional](https://swr.vercel.app/docs/conditional-fetching#conditional) so that the request never happens if a `locationUuid` isn't provided.